cli/src/semdep/parsers/util.py

```diff
@@ -257,12 +261,32 @@ def to_json(self) -> Dict[str, Union[Optional[str], Optional[int]]]:
         }
 
 
+@overload
 def safe_path_parse(
     path: Optional[Path],
     parser: "Parser[A]",
     parser_name: ParserName,
     preprocess: Callable[[str], str] = lambda ξ: ξ,  # ξ kinda looks like a string hehe
 ) -> Optional[A]:
+    ...
+
+
+@overload
+def safe_path_parse(
+    path: Optional[Path],
+    parser: Callable[[str], A],
+    parser_name: ParserName,
+    preprocess: Callable[[str], str] = lambda ξ: ξ,  # ξ kinda looks like a string hehe
+) -> Optional[A]:
+    ...
+
+
+def safe_path_parse(
+    path: Any,
+    parser: Any,
+    parser_name: Any,
+    preprocess: Any = lambda ξ: ξ,  # ξ kinda looks like a string hehe
+) -> Any:
```

Comment:
What is happening here?

Reply:
`@overload` lets you give two (non-overlapping) types to a function. I'm making it so `safe_path_parse` can be called either with a `Parser` object, or with just a `str -> A` function, so that I can pass in the `parse_yaml_preserve_spans` function. The actual definition just has `Any` because our pre-commit settings require type annotations on functions, using `@overload` means only the annotations with that decorator actually mean anything